resources/xsd/fastRTPS_profiles.xsd

```diff
@@ -5,293 +5,1124 @@
            attributeFormDefault="unqualified"
            elementFormDefault="qualified">
 
-    <xs:simpleType name="stringType">
-        <xs:restriction base="xs:string"/>
-    </xs:simpleType>
+    <!--| MAIN DOCUMENT HIERARCHY |-->
+    <!--__ROOT__:
+        ├ dds      [0~1],
+        |   ├ profiles [0~1],
+        |   ├ log      [0~1],
+        |   └ types    [0~1], (rooted definition)
+        |
+        ├ profiles [0~1],
+        ├ log      [0~1],
+        └ types    [0~1] (standalone definition) -->
+    <!-- dds, profiles, log or types can be the ROOT element of the profile -->
+    <xs:element name="dds">
+        <xs:complexType>
+            <xs:all>
+                <xs:element name="profiles" type="profilesType" minOccurs="0" maxOccurs="1"/>
+                <xs:element name="types" type="typesType" minOccurs="0" maxOccurs="1"/>
+                <xs:element name="log" type="logType" minOccurs="0" maxOccurs="1"/>
+            </xs:all>
+        </xs:complexType>
+    </xs:element>
+    <xs:element name="profiles" type="profilesType"/>
+    <xs:element name="types" type="typesType"/>
+    <xs:element name="log" type="logType"/>
+
+
+    <!--PROFILES:
+        ├ library_settings         [0~1],
+        ├ participant              [0~*],
+        ├ data_writer (=publisher) [0~*],
+        ├ data_reader (=subscriber)[0~*],
+        ├ topic                    [0~*],
+        └ transport_descriptors    [0~1] -->
+    <xs:complexType name="profilesType">
+        <xs:sequence minOccurs="0" maxOccurs="unbounded">   <!-- multiple instances of the elements -->
```

Comment:
Should we specify minOccur="1" ? Because an empty profiles element maybe doesn't make sense.
If so, we should do the same on the rest of the profiles.